### ISSUE TITLE
[TF 2.0] Support for TFGAN.

### ISSUE THREAD

**Issue by dynamicwebpaige** [tensorflow/tensorflow#26303]
[TFGAN](https://www.tensorflow.org/api_docs/python/tf/contrib/gan) is a popular lightweight library for training and evaluating GANs. In addition to providing the infrastructure for easily training and evaluating GANS, this library contains modules for a TFGAN-backed Estimator, evaluation metrics, features (such as virtual batch normalization), and losses.

The purpose of this feature request would be to migrate TFGAN to its own repo, and to ensure it is compatible with TensorFlow 2.0.

**Comment by Dexter2389** [comment]
@dynamicwebpaige I want to work on that!! How do I start??

**Comment by fenglui** [comment]
@Dexter2389 see https://github.com/zurutech/gans-from-theory-to-production

**Comment by galeone** [comment]
I'm currently migrating all notebooks (https://github.com/zurutech/gans-from-theory-to-production) to their Tensorflow 2.0 version.
Right now, without the new location of `tf.contrib.gan` the migration is stopped.

Thus @Dexter2389 I guess you can use notebooks 2.1, 3.0, and 3.1 as a reference.
In particular, in the 2.1 I already migrated the `discriminator_fn` to what I think the new implementation tf2 compatible (Keras based) should be. The other notebooks are just how to use tfgan.

When `tf.contrib.gan` is moved to its own repo and the models are migrated as in notebook 2.1 everything should work as in 1.x version.

**Comment by armando-fandango** [comment]
IF someone is willing to contribute and maintain, I propose that we move this to tf.addons for now.

**Comment by Dexter2389** [comment]
Ok!!

**Comment by armando-fandango** [comment]
@dynamicwebpaige Please let us know if there are any updates on this. I am starting to work on implementing GANS in our next project and would love to work on top of experimental 2.0 based release if possible.

**Comment by captain-pool** [comment]
> Ok!!

Hey @Dexter2389 what's the progress on this? I'ld like to collaborate.

**Comment by dynamicwebpaige** [comment]
TF-GAN is now compatible with TensorFlow 2.0, and can be found [here](https://github.com/tensorflow/gan). Let us know what you think! 😄

**Comment by RitikaAg** [comment]
@dynamicwebpaige so what would be the equivalent of tf.contrib.gan.eval.get_graph_def_from_disk('path to .pb file)
